src/shogun/mathematics/Statistics.h

```diff
@@ -44,6 +45,83 @@ class CStatistics: public CSGObject
 			return sum/vec.vlen;
 		}
 
+	/** Calculates equally weighted cumulative moving average.
+	 * Given \f$cma_{n-1}=\{av_1, ..., av_m\}\f$ and \f$datum=\{x_1, ..., x_m\}\f$,
+	 * this is \f$cma_n=\{av_1+\frac{x_1-av_1}{n}, ..., av_m+\frac{x_m-av_m}{n}\}\f$
+	 *
+	 * @param cma the previous moving average
+	 * @param datum new datum point
+	 * @param n number of data points including the new datum point
+	 * @param result the vector that saves the result 
+	 */
+	template<class T>
+		static void cumulative_moving_average(
+			const SGVector<T>& cma, const SGVector<T>& datum,
+			int32_t n, SGVector<T>& result) 
```

Comment:
Wouldn't it be better to return the result rather than modify the reference?

Reply:
Moving averages imply that something is modified so I think is good to just modify the given vector wih the new datum. Not even sure we need the result parameter. Would just call the method “update_mean” and then have only 3 parameters: the mean, the new vector, and n